src/snowflake/cli/api/config.py

```diff
@@ -47,6 +48,50 @@ class Empty:
 )
 
 
+@dataclass
+class ConnectionConfig:
+    account: Optional[str] = None
+    user: Optional[str] = None
+    password: Optional[str] = None
+    host: Optional[str] = None
+    region: Optional[str] = None
+    port: Optional[int] = None
+    database: Optional[str] = None
+    schema: Optional[str] = None
+    warehouse: Optional[str] = None
+    role: Optional[str] = None
+    authenticator: Optional[str] = None
+    private_key_path: Optional[str] = None
+
+    _other_settings: dict = field(default_factory=lambda: {})
+
+    @classmethod
+    def from_dict(cls, config_dict: dict) -> ConnectionConfig:
+        self = cls()
+        for key, value in config_dict.items():
+            if hasattr(self, key):
+                setattr(self, key, value)
+            else:
+                self._other_settings[key] = value
+        return self
```

Comment:
It's convention, but `self` is meant to represent instance, `config` or `connection_config` should be ok

Reply:
Changing this name causes our precommit hook to fail - probably only `self` instance is allowed to call private members and IMO we should be allowed to call them because even in classmethod, we're still in the context of class, especially in this factory method. I kept `self` as a var name.

```
ruff............................................................................Failed
- hook id: ruff
- exit code: 1

src/snowflake/cli/api/config.py:75:17: SLF001 Private member accessed: `_other_settings`
Found 1 error.
```

Reply:
I added one more commit changing `from_dict` function to use 2 dicts as we agreed.